charts/datadog/templates/daemonset.yaml

```diff
@@ -9,6 +9,9 @@ metadata:
   labels:
 {{ include "datadog.labels" . | indent 4 }}
     app.kubernetes.io/component: agent
+    {{- if .Values.clusterAgent.admissionController.enabled }}
```

Comment:
maybe we dont even needs the `{{- if ...}}` condition since it should have effect on the agent if the admission controller is not enabled